### DISCUSSION TITLE
How to manage memory buffer with Avalanche?

### DISCUSSION THREAD

**Discussion by bit0123** [ContinualAI/avalanche#432] [Q&A] [answered]
Dear Avalanche team,
Thanks for making this excellent prototyping tool for CL experiments.
I would appreciate if you please address my following questions:
1. How to split available Avalanche dataset into train and validation? Does standard approach for splitting pytorch Dataset work here?
2. How to manage memory buffer for keeping subset of samples from old task and marge that to new dataset for task at time t?

Thanks for your reply. 
Cheers!

**Comment by AntonioCarta** [comment, marked as answer]
Hello @bit0123,

1) you can't do it right now but we are going to solve it soon, probably next week. Follow issue #421 for updates on this.
2) We have `avalanche.training.plugins.ReplayPlugin` and related `avalanche.training.strategies.Replay`. You can use them directly or if you want to define your own custom replay you can start from their implementation.

**Reply by bit0123** [reply]
Hi @AntonioCarta , Thanks for prompt response.

**Reply by vlomonaco** [reply]
@bit0123 for 1. yes they would work but you won't be able to use the Avalanche _benchmarks generators_ on them till #421 is solved. Can we mark this question as solved? :-)

**Reply by bit0123** [reply]
@vlomonaco Sure, Please. Thanks!
